src/operators.jl

```diff
@@ -33,15 +35,9 @@ function map(f, input::Signal, inputsrest::Signal...;
 end
 
 function connect_map(f, output, inputs...)
-    let prev_timestep = 0
-        for inp in inputs
-            add_action!(inp, output) do output, timestep
-                if prev_timestep != timestep
-                    result = f(map(value, inputs)...)
-                    send_value!(output, result, timestep)
-                    prev_timestep = timestep
-                end
-            end
+    let
```

Comment:
I don't think so. Didn't want to mess with closures around the action functions where locals are defined, but I think you're right that they'd probably still work without the `let`s